sopel/plugins/rules.py

```diff
@@ -778,12 +782,69 @@ def kwargs_from_callable(cls, handler):
         }
 
     @classmethod
-    def from_callable(cls, settings, handler):
-        regexes = tuple(_clean_rules(
-            handler.rule,
+    def regex_from_callable(cls, settings, handler):
+        regexes = getattr(handler, cls.REGEX_ATTRIBUTE, []) or []
+        if not regexes:
+            raise RuntimeError(
+                'Invalid rule handler: %s' % handler)
+
+        return tuple(_clean_rules(
+            regexes,
             settings.core.nick,
             settings.core.alias_nicks,
         ))
+
+    @classmethod
+    def regex_from_callable_lazy(cls, settings, handler):
+        lazy_loaders = getattr(handler, cls.LAZY_ATTRIBUTE, [])
+        if not lazy_loaders:
+            raise RuntimeError(
+                'Invalid lazy rule: %s' % handler)
+
+        loader = tools.chain_loaders(*lazy_loaders)
+        regexes = loader(settings)
+
+        if not regexes:
+            raise RuntimeError(
+                'Invalid lazy loader: %s' % handler)
+
+        return regexes
+
+    @classmethod
+    def from_callable(cls, settings, handler):
+        regexes = cls.regex_from_callable(settings, handler)
+        kwargs = cls.kwargs_from_callable(handler)
+        kwargs['handler'] = handler
+
+        return cls(regexes, **kwargs)
+
+    @classmethod
+    def from_callable_lazy(cls, settings, handler):
+        """Instantiate a rule object from a handler with lazy-loaded regexes.
+
+        :param settings: Sopel's settings
+        :type settings: :class:`sopel.config.Config`
+        :param callable handler: a function-based rule handler with a
+                                 lazy-loader for the regexes
+        :return: an instance of this class created from the ``handler``
+        :rtype: :class:`AbstractRule`
+
+        Similar to the :meth:`from_callable` classmethod, it requires a rule
+        handlers decorated with :mod:`sopel.plugin`'s decorators.
```

Comment:
```suggestion
        handler decorated with :mod:`sopel.plugin`'s decorators.
```

I think, since there's only one `handler` arg.

Reply:
You never know with s, sometimes you may need one, sometimes not, it's so very confusing.